### DISCUSSION TITLE
iOS crash in iPhone 8 AGXA11Device

### DISCUSSION THREAD

**Discussion by AppHock** [Tencent/libpag#1881] [Q&A]
crasy log: [AGXA11Device originalObject]: unrecognized selector sent to instance 0x116858c00

**Comment by kevingpqi123** [comment]
这个 crash 和 pag 有啥关系，AGXA11Device 不是 pag 中的类

**Reply by AppHock** [reply]
感谢回复，我截图下堆栈信息，通过本地 bundle 加载的 pag 动画，时机比较早，在首页加载的

**Reply by AppHock** [reply]
![Pag 崩溃](https://github.com/Tencent/libpag/assets/14970137/68ff99c2-3f61-4fca-b8a1-32af5b873e08)

**Reply by kevingpqi123** [reply]
为了提高对接效果，请提供复现的 demo 工程，这样效率会更高些，我们没有时间给大家构造复现场景

**Reply by AppHock** [reply]
感谢朋友这么及时的回复，目前我们是用的版本是4.3.30。看调用栈是崩溃在Metal `_MTLCreateGLMetalDevice + 76这个里面，估计内部又去调用了AGXA11Device实例对象的 originalObject方法，方法列表里面又不存在这个方法，估计是 Apple 内部运行时根据设备类型和状态去主动调用了originalObject方法，没有实现导致崩溃，我这边已经在 Apple 论坛和stackoverflow上反馈了，看有没有大佬回复。

Apple forum: https://developer.apple.com/forums/thread/741734
stackoverflow: https://stackoverflow.com/questions/77505226/agxa11device-originalobject-crash-on-iphone-8

**Comment by kevingpqi123** [comment]
没有进一步的信息，暂时关闭 discussion，待有进一步信息时再打开
